explore/src/main/scala/explore/proposal/ProgramUsersTable.scala

```diff
@@ -150,6 +170,52 @@ object ProgramUsersTable:
             partnerSelector(pl, usersView.set)
       ),
       column(EmailColumnId, _.user.profile.foldMap(_.primaryEmail).getOrElse("-")),
+      ColDef(
+        ESColumnId,
+        _.educationalStatus,
+        enableSorting = true,
+        enableResizing = true,
+        cell = c =>
+
+          val cell   = c.row.original
+          val userId = cell.user.id
+          c.table.options.meta.map: meta =>
+            val view = meta.users
+              .zoom(esLens(userId))
+              .withOnMod(es =>
+                updateUserES[IO](meta.programId, userId, es.headOption.flatten).runAsyncAndForget
+              )
+
+            EnumOptionalDropdown[EducationalStatus](
+              id = "es".refined,
+              value = view.get.headOption.flatten,
+              showClear = true,
+              clazz = ExploreStyles.PartnerSelector,
+              onChange = es => view.set(es)
+            )
+      ),
+      ColDef(
+        ThesisColumnId,
+        _.thesis,
+        cell = c =>
+
+          val cell   = c.row.original
+          val userId = cell.user.id
+
+          c.table.options.meta.map: meta =>
+            val view = meta.users
+              .zoom(thesisLens(userId))
+              .withOnMod(th =>
+                updateUserThesis[IO](meta.programId,
+                                     userId,
+                                     th.headOption.flatten
+                ).runAsyncAndForget
```

Comment:
In other tables, we have turned a `View[List[Foo]]` to a `List[View[Foo]]` by using `toListOfViews`. Then each row is actually a `View`. Maybe that would simplify things here?